src/main/java/com/zaxxer/hikari/pool/PoolBase.java

```diff
@@ -332,17 +332,17 @@ Connection newConnection() throws Exception
     */
    private void setupConnection(final Connection connection) throws SQLException
    {
+      checkDriverSupport(connection);
```

Comment:
checkDriverSupport() may issue commit/rollback :) keep it after setting networkTimeout